service-front/module/Application/src/Helpers/FormProcessorHelper.php

```diff
@@ -34,10 +34,8 @@ public function processDrivingLicenceForm(
 
             $variables['dln_data'] = $formArray;
             $validDln = $this->opgApiService->checkDlnValidity($formArray['dln']);
-            $template = $validDln === 'PASS' ? $templates['success'] : $templates['fail'];
-            if ($validDln === 'PASS') {
-                $this->opgApiService->updateCaseSetDocumentComplete($uuid);
-            }
+
+            $template = $this->fraudCheck($validDln, $uuid, $templates);
```

Comment:
I have a slight concern that doing this inside the individual ID provider checks creates too much coupling and makes the code for processing someone's ID more complicated than necessary.

Another option would be that a successful result in any of these takes you to `/:uuid/fraud-check` which does the API call then redirects them to KBVs or a thin file error, which would keep the code for identity checking and fraud checking separate. But that's more a suggestion than a dealbreaker.

Reply:
I've pulled the fraud check back into the controller and it's all working ok, but it's increasingly revealing that the formProcessor, instead of being a useful way to abstract and reduce complexity, is mostly doing the opposite. I only introduced it to begin with because I didn't want fat and complicated controller methods, but it hasn't prevented that happening anyway.